docs/source/getting_started/AzureAuthentication.rst

```diff
@@ -0,0 +1,246 @@
+require authentication to use them. Some examples are:
+
+- Microsoft Sentinel, Microsoft Graph, Microsoft Defender and Azure
+  Resource Graph data providers
+- Microsoft Sentinel APIs
+- Azure Data APIs
+- Microsoft Sentinel TI provider
+
+MSTICPy uses Azure SDKs and standard Azure credentials to authenticate
+to these services.
+
+Azure supports many different types of credential - the different
+credential types are a function of the way that you authenticate to
+Azure Active Directory (AAD).
+
+Some credential types are distinguished based on what
+you use to authenticate your identity to AAD. For example:
+
+- **UsernamePasswordCredential** - simple username and password
+- **CertificateCredential** - certificate private key
+- **ClientSecretCredential** - AppID and App secret
+- **ManagedIdentityCredential** - Uses the AAD assigned identity for something
+  like a domain-joined Virtual machine.
+
+Other types have helper functionality that allow you to authenticate
+in specific circumstances:
+
+- **InteractiveBrowserCredential** - invokes a browser to initiate the
+  authentication flow.
+- **EnvironmentCredential** - looks for specific environment variables
+  that contain credential details such as a password or App secret.
+- **DeviceCodeCredential** - creates a one-time device code and returns
+  login URL to initiate the authentication flow.
+
+Some types of credential can be obtained from applications or system
+components in which you have previously authenticated
+such as:
+
+- **AzureCliCredential**
+- **VisualStudioCodeCredential**
+- **AzurePowerShellCredential**
+
+A relatively complete list can be found here
+`Azure Identity Python SDK Credential classes
+<https://learn.microsoft.com/python/api/overview/azure/identity-readme?view=azure-python#credential-classes>`_
+
+
+.. note:: In MSTICPy we refer to the different "credential types" as
+    "authentication methods". The function parameter used to specify
+    the method(s) that you want to use is ``auth_methods``.
+    The configuration item in msticpyconfig.yaml is also named
+    ``auth_methods`` and is in the Azure section of the settings.
+
+
+How to choose a credential type
+-------------------------------
+Since there are so many credential types, working out which one to use
+can be confusing.
+
+MSTICPy, by default, will try to use the following credential types (and in this
+order):
+
+- Managed Identity
+- Azure CLI
+- Device Code
+
+This means that MSTICPy will try to obtain credentials of each these
+types in turn. If it finds a matching credential type it will try
+to use that to authenticate you to the Azure service that you are
+trying to access. In the default case it will do the following:
+
+1. Check to see if you are running on a system (a host/virtual machine)
+   that has a system-assigned identity OR (as in the case of Azure
+   Machine Learning) has support for user-assigned managed identities.
```

Comment:
Do we want to link to documents on Azure Managed Identities here? Its not something people are often familiar with?